sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/ContainerRegistryRedirectPolicy.java

```diff
@@ -0,0 +1,139 @@
+                    .then(this.attemptRedirect(context, next, redirectRequestCopy, redirectAttempt + 1, attemptedRedirectUrls))
+                    .flatMap(newResponse -> {
+                        String digest = httpResponse.getHeaders().getValue(DOCKER_DIGEST_HEADER_NAME);
+                        if (digest != null) {
+                            newResponse.getHeaders().add(DOCKER_DIGEST_HEADER_NAME, digest);
+                        }
+                        return Mono.just(newResponse);
+                    });
+            } else {
+                return Mono.just(httpResponse);
+            }
+        });
+    }
+
+    public boolean shouldAttemptRedirect(HttpPipelineCallContext context, HttpResponse httpResponse, int tryCount, Set<String> attemptedRedirectUrls) {
+        if (this.isValidRedirectStatusCode(httpResponse.getStatusCode()) && this.isValidRedirectCount(tryCount) && this.isAllowedRedirectMethod(httpResponse.getRequest().getHttpMethod())) {
+            String redirectUrl = this.tryGetRedirectHeader(httpResponse.getHeaders(), REDIRECT_LOCATION_HEADER_NAME);
+            if (redirectUrl != null && !this.alreadyAttemptedRedirectUrl(redirectUrl, attemptedRedirectUrls)) {
+                LOGGER.verbose("[Redirecting] Try count: {}, Attempted Redirect URLs: {}", tryCount, String.join(",", attemptedRedirectUrls));
+                attemptedRedirectUrls.add(redirectUrl);
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    private HttpRequest createRedirectRequest(HttpResponse httpResponse) {
+        String responseLocation = this.tryGetRedirectHeader(httpResponse.getHeaders(), REDIRECT_LOCATION_HEADER_NAME);
+        HttpRequest request = httpResponse.getRequest();
+        request.setUrl(responseLocation);
+        request.getHeaders().remove(AUTHORIZATION);
+        return httpResponse.getRequest().setUrl(responseLocation);
+    }
+
+    private boolean alreadyAttemptedRedirectUrl(String redirectUrl, Set<String> attemptedRedirectUrls) {
+        if (attemptedRedirectUrls.contains(redirectUrl)) {
+            LOGGER.error("Request was redirected more than once to: {}", new Object[]{redirectUrl});
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean isValidRedirectCount(int tryCount) {
+        if (tryCount >= MAX_REDIRECT_ATTEMPTS) {
+            LOGGER.error("Request has been redirected more than {} times.", new Object[]{MAX_REDIRECT_ATTEMPTS});
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    private boolean isAllowedRedirectMethod(HttpMethod httpMethod) {
+        if (REDIRECT_ALLOWED_METHODS.contains(httpMethod)) {
+            return true;
+        } else {
+            LOGGER.error("Request was redirected from an invalid redirect allowed method: {}", new Object[]{httpMethod});
```

Comment:
I think `new Object[] {httpMethod}`  will log the reference of the array instance and will not be very useful for troubleshooting. Why do we need the array? Can we just log the `httpMethod`?